Clean up test suite scripts

Naufragous opened the pull request:
The test scripts have grown to be quite big (over 15000 lines) and
are hard to navigate. There are multiple if branches with over
40 conditional checks chained together. This commit solves some of
those issues.

 - Unite big repetetive if conditions into clean array lookups
 - Move 'install help' commands to a separate shell script
 - Adjust array lookup in test.sh to behave more intuitive
 - Add comments at key points to simplify navigation
 - Code formatting

jsteube commented:
- Can you please replace the smart match operator with a normal function which takes the array and the keyword instead, analogue to is_in_array() in the shellscript?
- Is the pygost working with pip3 as well? In that case, everything is OK. Otherwise please replace the "pip" with "pip2" in the depency install shellscript
- OCD error:
``` 
VC_MODES="    13711 13712 13713 13721 13722 13723 13731 13732 13733 13751 13752\
  13753 13771 13772 13773"
```

Naufragous commented:
> replace the smart match operator with a normal function

Done.
> Is the pygost working with pip3 as well?

It is, but since the python code is executed with `python2 -c`, pip2 should be used indeed. Done.

> OCD error

Done :)

Naufragous committed:
Clean up test suite scripts

The test scripts have grown to be quite big (over 15000 lines) and
are hard to navigate. There are multiple if branches with over
40 conditional checks chained together. This commit solves some of
those issues.

 - Unite big repetetive if conditions into clean array lookups
 - Move 'install help' commands to a separate shell script
 - Adjust array lookup in test.sh to behave more intuitive
 - Add comments at key points to simplify navigation
 - Code formatting
